.github/workflows/main.yml

```diff
@@ -53,22 +53,26 @@ jobs:
   standalone_spark3_with_S3:
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        python-version: [ 3.9, 3.11 ]
+
     steps:
     - uses: actions/checkout@v2
     
     - name: Build spark-docker
-      run: docker build -t spark-docker ./examples/spark-with-S3 --build-arg SPARK_INPUT_VERSION=3.2.2 --build-arg PYTHON_VERSION=3.9.15
+      run: docker build -t spark-docker ./examples/spark-with-S3 --build-arg SPARK_INPUT_VERSION=3.2.2 --build-arg PYTHON_VERSION=${{ matrix.python-version }}
     
     - name: Build the docker-compose stack
       run: |
-        export PYTHON_VERSION=3.9
+        export PYTHON_VERSION=${{ matrix.python-version }}
         docker compose -f ./examples/spark-with-S3/docker-compose.yml up -d
       
     - name: Check running containers
       run: docker ps -a
 
     - name: Run spark Job
-      run: docker exec spark-master ./examples/spark-with-S3/scripts/run_spark_example.sh python3.9 3.2.2
+      run: docker exec spark-master ./examples/spark-with-S3/scripts/run_spark_example.sh python${{ matrix.python-version }} 3.2.2
```

Comment:
It look like you need spark >= 3.4 in this case
https://github.com/apache/spark/pull/38987